plover/gui_qt/dictionaries_widget.py

```diff
@@ -348,4 +348,15 @@ def _save_as(self):
                     dictionaries = list(self._dictionaries)
                     if new_filename not in self._dictionaries:
                         dictionaries.append(new_filename)
-                    self._update_dictionaries(dictionaries)
\ No newline at end of file
+                    self._update_dictionaries(dictionaries)
+
+                entries = result.save.error_msg
```

Comment:
1. Placing this code here is troublesome, because if the conversion fails, `result` could be None.

I'd throw it into the `else` clause.

```python
try:
    things
except:
    errors
else:
    if result is not None and result.save.error_msg:
        msg = _('blah blah').format(stuff)
        QMessageBox.information(...)
```

2. It is strange to me, too, that the errors show *after* it asks if you want to add it to your existing list.
3. Maybe the QMessageBox isn't the best idea. If there are hundreds of errors, it would extend off screen. Ideally we would use a dialog that's just a simple text field that is scrollable and copy-able (for users who want to copy-paste the output into a help forum or the like).

To do this, it would not be hard to create a QDialog subclass and just call MyCustomDialog(msg, self).exec()

Overall, pretty impressive.